src/components/clusterDetail/ClusterDetail.tsx

```diff
@@ -30,35 +30,25 @@ import { AlertsContext } from '../AlertsContextProvider';
 import { canDownloadClusterLogs } from '../hosts/utils';
 
 const canAbortInstallation = (cluster: Cluster) => {
-  if (
-    !['preparing-for-installation', 'installing', 'installing-in-progress'].includes(cluster.status)
-  ) {
-    return false;
-  }
-
-  if (cluster.hosts) {
-    if (cluster.hosts.find((h) => h.status === 'installing-pending-user-action')) {
-      // a host in installing-pending-user-action
-      return false;
-    }
-
-    if (cluster.hosts.find((h) => h.status === 'error')) {
-      // a host is in error
-      if (cluster.hosts.find((h) => h.status !== 'installing')) {
-        // no host is in installing
-        if (
-          cluster.hosts.find(
-            (h) => !['error', 'installed', 'preparing-for-installation'].includes(h.status),
-          )
-        ) {
-          // a host not yet in error or installed or preparing-for-installation
-          return false;
-        }
-      }
-    }
-  }
-
-  return true;
+  const allowedClusterStates = [
+    'preparing-for-installation',
+    'installing',
+    'installing-in-progress',
+    'finalizing',
+  ];
+  const allowedHostStates = [
```

Comment:
What about `resetting`?
I think that `pending-user-action` is `installing-pending-user-action`

Reply:
`resetting` is done after the installation is cancelled so I don't think it makes sense to add it as a state from which we can cancel.

fixed to 'installing-pending-user-action' and added type to the arrays to avoid errors such as this.